File: 251229/키를 기준으로 정렬/sort-by-height.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);

class Person {
    constructor(name, height, weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
}

const people = []
for (let i = 1; i <= n; i++) {
    const [name, height, weight] = input[i].split(' ');
    people.push(new Person(name, Number(height), Number(weight)));
}

people.sort((a, b) => a.height - b.height);

people.forEach(person => {
    console.log(`${person.name} ${person.height} ${person.weight}`);
})

File: 251229/키를 기준으로 정렬/sort-by-height.py
class Person:
    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight

n = int(input())
people = []
for _ in range(n):
    n, h, w = input().split()
    people.append(Person(n, int(h), int(w)))

people.sort(key=lambda x: x.height)

for person in people:
    print(person.name, person.height, person.weight)